ndinterp/src/grid/cubic.rs

```diff
@@ -18,40 +19,35 @@ pub use crate::interpolate::Interpolator;
 /// with hij the Hermite basis functions
 #[derive(Debug)]
 pub struct Cubic1d {
-    pub grid: Grid,
+    /// The grid object contains all necessary information to perform the interpolation
+    pub grid: Grid<Ix1>,
 }
 
-impl Interpolator<f64> for Cubic1d {
-    /// Use Cubic interpolation 1d to compute y(query)
-    /// The interpolation uses the two nearest neighbours and their derivatives computed as an
-    /// average of the differences above and below.
-    ///
-    /// Special cases are considered when the interpolation occurs between the first (last) two
-    /// bins, where the derivative would involve points outside the grids.
-    ///
-    /// Two special are considered, when the interpolation occurs between the first (last) two
-    /// bins, the derivative at the boundary is approximated by the forward (backward) difference
-    fn interpolate(&self, query: f64) -> Result<f64, InterpolationError> {
-        let idx = self.grid.closest_below(query)?;
-        let dx = self.grid.input[idx + 1] - self.grid.input[idx];
+impl<'a> GridSlice<'a> {
+    /// Implements utilities for a GridSlice that can be used by cubic interpolation Nd
+    /// Takes as input the value being queried and its index within the given slice
+    fn cubic_interpolate_1d(&'a self, query: f64, idx: usize) -> Result<f64, InterpolationError> {
```

Comment:
Does it make sense to call it `1d`? A slice is always 1D... (or are you already planning to extend beyond?)

Reply:
Who knows… might be necessary in the future.